### PR TITLE
greetd: init at 0.7.0

### PR DESCRIPTION
**Pull request by luc65r**
###### Motivation for this change
Wanted to package greetd and friends

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by luc65r** [comment]
Related: #102225 
I will do a module later, I'll need some help

**Comment by luc65r** [comment]
Should I move the packages to `pkgs/applications/display-managers/`?
greetd isn't really a display manager, but it's close...

**Comment by tsujp** [comment]
Argh you beat me to `greetd`!

WRT 

> Should I move the packages to `pkgs/applications/display-managers/`?
> greetd isn't really a display manager, but it's close...

It does manage the _actual_ session managers and so is a part of that system, I'd say it's fine to put it in `display-managers` (that's where I had it in my implemented nix package). If you need any assistance let me know (I'm ultra new to Nix so I think you've definitely got this) and I'll see if I can assist :)

**Review by mkg20001 — CHANGES_REQUESTED** [review]
needs https://github.com/NixOS/nixpkgs/pull/102225 merged, mobule possibly needs to be moved (where do wayland based DMs go?)

**Comment by qzle** [comment]
> I will do a module later, I'll need some help

@luc65r I have a module ready to go: [greetd.nix](https://git.c3pb.de/jens/dotfiles/-/blob/master/modules/greetd.nix).
Would it be ok for you if I create a PR for that?


> [...] mobule possibly needs to be moved (where do wayland based DMs go?)

Technically greetd is not wayland-based and makes no assumptions about either the greeter or the session (relevant quote from the docs: "If you can run it from your shell in a TTY, greetd can start it").

That doesn't really make classification easier though. I guess `display-managers` is fine? There is a hard dependency on PAM, so `os-specific/linux` would also be correct.

**Comment by luc65r** [comment]
> Would it be ok for you if I create a PR for that?

@qzle Sure!
[I wrote something](https://github.com/luc65r/nixpkgs/blob/nixos/greetd/nixos/modules/services/wayland/greetd.nix), but I've been too lazy to test it.

> That doesn't really make classification easier though. I guess `display-managers` is fine? There is a hard dependency on PAM, so `os-specific/linux` would also be correct.

That was my reasoning, I guess it doesn't really matter where it is.

**Comment by cidkidnix** [comment]
If possible and agreeable I would put all of the greeters (gtkgreet, wlgreet, etc..) under the `greetd.` prefix so it the greeters would just be `greetd.gtkgreet` or `greetd.wlgreet`, as imo it makes it cleaner in the long run

**Comment by devhell** [comment]
Aside from the conflict, what else is stopping this from getting merged?

**Comment by luc65r** [comment]
> I would put all of the greeters (gtkgreet, wlgreet, etc..) under the `greetd.` prefix

Great idea! I will do it as soon as possible.

> what else is stopping this from getting merged?

After that, nothing.

File: pkgs/os-specific/linux/gtkgreet/default.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    gtk-layer-shell
+    json_c
+    scdoc
+  ];
+
+  mesonFlags = [
+    "-Dlayershell=enabled"
+  ];
+
+  meta = with stdenv.lib; {
```
Comment (fortuneteller2k) [inline pkgs/os-specific/linux/gtkgreet/default.nix:41]:
```suggestion
  meta = with lib; {
```

File: pkgs/os-specific/linux/gtkgreet/default.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
+{ stdenv
```
Comment (fortuneteller2k) [inline pkgs/os-specific/linux/gtkgreet/default.nix:1]:
```suggestion
{ stdenv
```
```suggestion
{ lib
, stdenv
```

File: pkgs/os-specific/linux/gtkgreet/default.nix (hunk 1 above)
Comment (fortuneteller2k) [reply pkgs/os-specific/linux/gtkgreet/default.nix:41]: use `lib` instead of `stdenv.lib`

File: pkgs/os-specific/linux/gtkgreet/default.nix (hunk 2 above)
Comment (fortuneteller2k) [reply pkgs/os-specific/linux/gtkgreet/default.nix:1]: see other comment below

**Review by moni-dz — CHANGES_REQUESTED** [review]

File: pkgs/os-specific/linux/gtkgreet/default.nix (hunk 1 above)
Comment (fortuneteller2k) [inline pkgs/os-specific/linux/gtkgreet/default.nix:41]:
```suggestion
  meta = with lib; {
```

Comment (fortuneteller2k) [reply pkgs/os-specific/linux/gtkgreet/default.nix:41]: use `lib` instead

File: pkgs/os-specific/linux/greetd/default.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  ];
+
+  buildInputs = [
+    pam
+  ];
+
+  postInstall = ''
+    for f in man/*; do
+      scdoc < "$f" > "$(sed 's/-\([0-9]\)\.scd$/.\1/' <<< "$f")"
+      rm -f "$f"
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/greetd/default.nix:34]:
```suggestion
      rm "$f"
```

File: pkgs/os-specific/linux/gtkgreet/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/os-specific/linux/gtkgreet/default.nix:41]: You must use lib or ofborg fails.

File: pkgs/os-specific/linux/gtkgreet/default.nix [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  src = fetchFromSourcehut {
+    owner = "~kennylevinsen";
+    repo = pname;
+    rev = version;
+    sha256 = "ms+2FdtzzNlmlzNxFhu4cpX5H+5H+9ZOtZ0p8uVA3lo=";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+    meson ninja
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/gtkgreet/default.nix:26]:
```suggestion
    meson
    ninja
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by devhell** [comment]
Wow, @luc65r, that's awesome stuff! Especially adding all those greeters! Fantastic, thank you! :smile:

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 102242` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
5 packages built:
</summary>
<ul>
<li>greetd.dlm</li>
<li>greetd.greetd</li>
<li>greetd.gtkgreet</li>
<li>greetd.tuigreet</li>
<li>greetd.wlgreet</li>
</ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]
LGTM

**Comment by lovesegfault** [comment]
This is awesome @luc65r, looking forward to the modules landing!

**Comment by devhell** [comment]
I second @lovesegfault, this is awesome. Thank you so much!

**Pull request merged**